common/timedcache/timedcache.go

```diff
@@ -84,13 +96,17 @@ func (tc *TimedCache) Get(key interface{}) (value interface{}, ok bool) {
 // Contains checks if a key is in the cache, without updating the
 // recent-ness or deleting it for being stale.
 func (tc *TimedCache) Contains(key interface{}) bool {
+	tc.lock.Lock()
+	defer tc.lock.Unlock()
 	_, ok := tc.Peek(key)
```

Comment:
Peek will grab the lock as well, probably deadlock

Reply:
You probably meant `tc.cache.Peek(key)`

Reply:
good catch. Fixed.

I do still want to use the `tc.Peek()`, because that will trigger a time based expiration, but I just don't need to lock the mutex